safe/impact_functions/inundation/flood_raster_road/impact_function.py

```diff
@@ -0,0 +1,394 @@
+# coding=utf-8
+"""Impact of flood on roads."""
+from qgis.core import (
+    QGis,
+    QgsCoordinateReferenceSystem,
+    QgsCoordinateTransform,
+    QgsFeature,
+    QgsFeatureRequest,
+    QgsField,
+    QgsGeometry,
+    QgsPoint,
+    QgsRectangle,
+    QgsSpatialIndex,
+    QgsVectorFileWriter,
+    QgsVectorLayer
+)
+from PyQt4.QtCore import QVariant
+
+from safe.impact_functions.base import ImpactFunction
+from safe.common.tables import Table, TableRow
+from safe.impact_functions.inundation.flood_raster_road\
+    .metadata_definitions import FloodRasterRoadsMetadata
+from safe.utilities.i18n import tr
+from safe.storage.vector import Vector
+from safe.common.utilities import get_utm_epsg, unique_filename
+from safe.common.exceptions import GetDataError
+from safe.gis.qgis_raster_tools import clip_raster
+from safe.gis.qgis_vector_tools import (
+    extent_to_geo_array,
+    create_layer)
+
+
+def _raster_to_vector_cells(raster, threshold_min, threshold_max, roads_crs):
```

Comment:
Can you make a unit test for this please? I also think we should move this to one of our utility classes e.g. safe/gis/qgis_vector_tools.py